executor/src/witgen/mod.rs

```diff
@@ -394,11 +428,13 @@ impl<'a, T: FieldElement> FixedData<'a, T> {
     }
 
     fn external_witness(&self, row: DegreeType, column: &PolyID) -> Option<T> {
-        let row = row % self.degree;
         self.witness_cols[column]
             .external_values
             .as_ref()
-            .and_then(|v| v.get(row as usize).cloned())
+            .and_then(|v| {
+                let row = row % v.len() as u64;
```

Comment:
Is this really the right thing to do? Shouldn't we have done that before already?

Reply:
Ah sorry, it was done before in line 397!